### PR TITLE
Feature/change page h1 on byo

### PR DESCRIPTION
**Pull request by thomasridd**
### What

Added IsCustom to the page model and page titles to the asset files
Page H1 and title customised for BYO datasets
Rippled through changes to the tests

image

image

### How to review

Media review
Tests pass
Sense check

### Who can review

Frontend go dev

### CODE REVIEW

File: handlers/overview.go [hunk 1]
```diff
@@ -368,8 +369,10 @@ func filterFlexOverview(w http.ResponseWriter, req *http.Request, f *FilterFlex,
 	}
 
 	basePage := f.Render.NewBasePageModel()
+	custom := helpers.IsBoolPtr(filterJob.Custom)
```
Comment (mr-nick17) [inline handlers/overview.go:372]: Is this needed here?

File: handlers/overview.go [hunk 2]
```diff
@@ -368,8 +369,10 @@ func filterFlexOverview(w http.ResponseWriter, req *http.Request, f *FilterFlex,
...
+
 	m := mapper.NewMapper(req, basePage, eb, lang, serviceMsg, filterID)
-	overview := m.CreateFilterFlexOverview(*filterJob, fDims, dimDescriptions, pops, *sdc, isMultivariate)
+	overview := m.CreateFilterFlexOverview(*filterJob, fDims, dimDescriptions, pops, *sdc, isMultivariate, custom)
```
Comment (mr-nick17) [inline handlers/overview.go:375]: The `filterJob` struct is already a parameter
```suggestion
	overview := m.CreateFilterFlexOverview(*filterJob, fDims, dimDescriptions, pops, *sdc, isMultivariate)
```

File: mapper/overview.go [hunk 3]
```diff
@@ -16,17 +16,24 @@ import (
...
 	queryStrValues := m.req.URL.Query()["showAll"]
 	path := m.req.URL.Path
 
 	p := model.Overview{
 		Page: m.basePage,
 	}
-	mapCommonProps(m.req, &p.Page, reviewPageType, "Review changes", m.lang, m.serviceMsg, m.eb)
+
+	title := helper.Localise("OverviewTitle", m.lang, 1)
+	if isCustom {
```
Comment (mr-nick17) [inline mapper/overview.go:28]:
```suggestion
	if helpers.IsBoolPtr(filterJob.Custom) {
```

File: model/overview.go [hunk 4]
```diff
@@ -13,6 +13,7 @@ type Overview struct {
 	IsMultivariate        bool        `json:"is_multivariate"`
 	EnableGetData         bool        `json:"enable_get_data"`
 	HasSDC                bool        `json:"has_sdc"`
+	IsCustom              bool        `json:"is_custom"`
```
Comment (mr-nick17) [inline model/overview.go:16]: Is this `bool` used anywhere?

File: mapper/overview_test.go [hunk 5]
```diff
@@ -200,10 +200,18 @@ func TestOverview(t *testing.T) {
...
+
+	Convey("test filter flex overview maps for custom datasets correctly", t, func() {
+		isCustom := true
+		overview := m.CreateFilterFlexOverview(filterJob, filterDims, dimDescriptions, pops, sdc, false, isCustom)
+		So(overview.Metadata.Title, ShouldEqual, "Create a custom dataset")
 	})
 
 	Convey("test truncation maps as expected", t, func() {
-		overview := m.CreateFilterFlexOverview(filterJob, filterDims, dimDescriptions, pops, sdc, false)
+		overview := m.CreateFilterFlexOverview(filterJob, filterDims, dimDescriptions, pops, sdc, false, false)
```
Comment (mr-nick17) [inline mapper/overview_test.go:214]: Moving the custom conditional to the mapper will save updating all of these tests

**Review by mr-nick17 — COMMENTED** [review]
Some questions/comments. The concourse jobs also look like they're failing too 😭

File: assets/locales/service.en.toml [hunk 6]
```diff
@@ -283,3 +283,11 @@ other = "<br><a href=\"/peoplepopulationandcommunity/populationandmigration/popu
...
 one = "<strong>{{.arg0}} area available</strong>"
 other = "<strong>All {{.arg0}} areas available</strong>"
+
+[OverviewTitle]
+description = "Overview page title"
+one = "Review changes"
+
+[OverviewCustomTitle]
+description = "Overview page title for Create a custom dataset"
+one = "Create a custom dataset"
```
Comment (mr-nick17) [inline assets/locales/service.en.toml:293]: Custom dataset is on the ticket/figma journey

File: handlers/overview.go (hunk 1 above)
Comment (thomasridd) [reply handlers/overview.go:372]: It was for some reason - not sure why but I had to wade an ocean of logging to figure it

**Comment by thomasridd** [comment]
You're right about all of this - I think I'll scale right back and start from scratch

**Pull request closed without merging**